src/request.ts

```diff
@@ -36,17 +36,27 @@ export interface IPrepareRequestLinkFulfillmentTransactionProps {
 	tokenDecimals: number
 }
 
-export interface IPrepareXchainRequestFulfillmentTransactionProps {
+export type IPrepareXchainRequestFulfillmentTransactionProps = {
 	senderAddress: string
 	fromToken: string
 	fromTokenDecimals: number
 	fromChainId: string
-	link: string
 	squidRouterUrl: string
 	provider: ethers.providers.Provider
-	apiUrl?: string
 	tokenType: EPeanutLinkType
-}
+} & (
+	| {
+			link: string
+			apiUrl?: string
+			APIKey?: string
+	  }
+	| {
+			linkDetails: Pick<
+				IGetRequestLinkDetailsResponse,
+				'chainId' | 'recipientAddress' | 'tokenAmount' | 'tokenDecimals' | 'tokenAddress'
+			>
```

Comment:
Thought hmm this is cool but do we need it? seems overly complex
Maybe can add comment as to why it's a good idea in the code?

Reply:
a comment would have been good. Its to support the pass details vs fetch details. Maybe having those types named would help